src/features/transfer-ownership/TransferOwnershipForm/TransferOwnershipForm.test.tsx

```diff
@@ -0,0 +1,465 @@
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+
+		test('should display error message when input value is equal to connected account address', async () => {
+			testSetup();
+
+			// input error message
+			const errorMessage = 'The address entered already owns this domain';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: mock.connectedAccount } });
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+
+		test('should display error message when input value is equal to connected account address', async () => {
+			testSetup();
+
+			// input error message
+			const errorMessage = 'The address entered already owns this domain';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: mock.connectedAccount } });
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
+
+		test('should display error message when connected account is not domain owner(0xxx)', async () => {
+			// edit mock.domain.owner
+			const mockOwner = '0xxx';
+
+			renderComponent(mockOwner);
+
+			// input error message
+			const errorMessage = 'You are not the owner of this domain';
+
+			expect(screen.queryByText(errorMessage)).not.toBeInTheDocument();
+
+			const onConfirmInputButton = await screen.findByRole('button', {
+				name: /transfer/i,
+			});
+
+			expect(onConfirmInputButton).toBeInTheDocument();
+			expect(onConfirmInputButton).toHaveAttribute('aria-disabled', 'true');
+
+			const input = await screen.findByPlaceholderText('Ethereum Wallet');
+
+			expect(input).toBeInTheDocument();
+
+			await act(async () => {
+				fireEvent.input(input, { target: { value: mock.validInputAddress } });
+			});
+
+			expect(onConfirmInputButton).not.toHaveAttribute('aria-disabled');
+
+			fireEvent.click(onConfirmInputButton);
+
+			await screen.findByText(errorMessage);
+		});
```

Comment:
This is probably more of a feedback item for the Design team, but I think the modal should just say "You are not the owner of this domain" from the start if you're not the owner, rather than waiting until submit.

Reply:
Yeah nice suggestion. This was actually something I added myself as a temp measure to handle the scenario. Shall hit Mike/Eth up